internal/provider/vaultsecrets/resource_vault_secrets_integration.go

```diff
@@ -0,0 +1,723 @@
+	AwsFederatedWorkloadIdentity     types.Object `tfsdk:"aws_federated_workload_identity"`
+	AzureClientSecret                types.Object `tfsdk:"azure_client_secret"`
+	AzureFederatedWorkloadIdentity   types.Object `tfsdk:"azure_federated_workload_identity"`
+	ConfluentStaticCredentialDetails types.Object `tfsdk:"confluent_static_credentials"`
+	GcpServiceAccountKey             types.Object `tfsdk:"gcp_service_account_key"`
+	GcpFederatedWorkloadIdentity     types.Object `tfsdk:"gcp_federated_workload_identity"`
+	MongoDBAtlasStaticCredentials    types.Object `tfsdk:"mongodb_atlas_static_credentials"`
+	TwilioStaticCredentials          types.Object `tfsdk:"twilio_static_credentials"`
+
+	// Computed fields
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ResourceID     types.String `tfsdk:"resource_id"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+
+	// Inner API-compatible models derived from the Terraform fields
+	capabilities                   []*secretmodels.Secrets20231128Capability                          `tfsdk:"-"`
+	awsAccessKeys                  *secretmodels.Secrets20231128AwsAccessKeysRequest                  `tfsdk:"-"`
+	awsFederatedWorkloadIdentity   *secretmodels.Secrets20231128AwsFederatedWorkloadIdentityRequest   `tfsdk:"-"`
+	azureClientSecret              *secretmodels.Secrets20231128AzureClientSecretRequest              `tfsdk:"-"`
+	azureFederatedWorkloadIdentity *secretmodels.Secrets20231128AzureFederatedWorkloadIdentityRequest `tfsdk:"-"`
+	confluentStaticCredentials     *secretmodels.Secrets20231128ConfluentStaticCredentialsRequest     `tfsdk:"-"`
+	gcpServiceAccountKey           *secretmodels.Secrets20231128GcpServiceAccountKeyRequest           `tfsdk:"-"`
+	gcpFederatedWorkloadIdentity   *secretmodels.Secrets20231128GcpFederatedWorkloadIdentityRequest   `tfsdk:"-"`
+	mongoDBAtlasStaticCredentials  *secretmodels.Secrets20231128MongoDBAtlasStaticCredentialsRequest  `tfsdk:"-"`
+	twilioStaticCredentials        *secretmodels.Secrets20231128TwilioStaticCredentialsRequest        `tfsdk:"-"`
+}
+
+var _ resource.Resource = &resourceVaultSecretsIntegration{}
+var _ resource.ResourceWithConfigure = &resourceVaultSecretsIntegration{}
+var _ resource.ResourceWithModifyPlan = &resourceVaultSecretsIntegration{}
+var _ resource.ResourceWithImportState = &resourceVaultSecretsIntegration{}
+
+func NewVaultSecretsIntegrationResource() resource.Resource {
+	return &resourceVaultSecretsIntegration{}
+}
+
+type resourceVaultSecretsIntegration struct {
+	client *clients.Client
+}
+
+func (r *resourceVaultSecretsIntegration) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_secrets_integration"
+}
+
+func (r *resourceVaultSecretsIntegration) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := map[string]schema.Attribute{
+		"provider_type": schema.StringAttribute{
```

Comment:
provider_type was used instead of provider since it is a reserved Terraform word.